packages/libraries/client/src/client.ts

```diff
@@ -36,6 +36,18 @@ export function createHive(options: HivePluginOptions): HiveClient {
     await Promise.all([schemaReporter.dispose(), usage.dispose()]);
   }
 
+  if (options.autoDispose) {
+    const signals = Array.isArray(options.autoDispose)
+      ? options.autoDispose
+      : ['SIGINT', 'SIGTERM'];
+    for (const signal of signals) {
+      process.once(signal, async () => {
+        logger.info(`[hive] Received ${signal}, disposing.`);
+        await dispose();
+      });
+    }
```

Comment:
Let's check first if `process` is even available (non-node envs), if it's not available, let's print a warning maybe?

Reply:
Hummm didn't though about that :/
I'm not very comfortable with the idea of having env specific behaviours like this.

Is it actually a warning ? Because we can't actually detect if the user has made a proper dispose setup or not. Would be terrible to always show the warning even if the user has actually handled this properly ^^'

Perhaps we should not do this and simply make it clear in the documentation that `dispose` *should* be called.

Reply:
This code runs when autodispose === true, so the intent is to auto dispose the client, but if a user tries to do something that is not possible, let's tell him/her about it.

Reply:
Yes, but `autodispose` is true by default for now.
Should I make it disabled by default ?
Or enabled by default only of `process` exists ?

Reply:
Maybe we should add "to disable this warning, disable autodispo" in the warning messge?

Reply:
Maybe something like:

> It seems that GraphQL Hive is not being executed in Node.js, which is why the `autodispose` feature is not working as intended. Please attempt manual client disposal and use `autodispose: false`.

Reply:
this way we tell user that `autodispose` is ineffective and there's a chance we notified him/her about an issue, but if the user is handling it manually anyway, it explains how to get rid of the warning.

Reply:
Should it be a warning then ? It seems it probably should be an exception at this point ?

Reply:
exception = user's app is broken and it'd be our fault